### PR TITLE
17 create expense claim

### PR DESCRIPTION
**Pull request by ramish94**
Added UI and code for creating expense claim.

Also added testing file for creating expense claim.

#17

### CODE REVIEW

File: ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/AddExpenseClaimActivity.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import android.os.Bundle;
+import android.text.InputType;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.widget.DatePicker;
+import android.widget.EditText;
+
+public class AddExpenseClaimActivity extends Activity implements OnClickListener {
```
Comment (adamjford) [inline ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/AddExpenseClaimActivity.java]: Think we need to have a Javadoc comment here, right?

Comment (ramish94) [inline ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/AddExpenseClaimActivity.java]: Javadoc comment for what?

Comment (adamjford) [inline ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/AddExpenseClaimActivity.java]: A Javadoc comment at the top of every class describing what it is, what it does, etc.

Comment (adamjford) [inline ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/AddExpenseClaimActivity.java]: From Project Part 4:

>For each source code file, you should have a brief introductory comment describing its purpose or role within the application or a design pattern, as well as any currently outstanding issues.

Comment (ramish94) [inline ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/AddExpenseClaimActivity.java]: Ah. alright. Mind must've glossed over that. I'll do that.

**Comment by ItsTristan** [comment]
Your tests aren't really doing anything useful. Can you make some tests that check that changes to the ExpenseClaim object's data is reflected in the TextViews' data and vice versa?

**Comment by saemorris** [comment]
So I realize that our UML doesn't reflect this, but won't an ExpenseClaim need an array of ExpenseItems? Or some sort of an ExpenseItemList?

**Comment by adamjford** [comment]
Wow, how did we miss that?

**Comment by ItsTristan** [comment]
That was my bad. It was in the UML, but isn't in the ExpenseClaim code. Regardless, it won't affect this particular issue; iirc, adding expense items is done from the summary page anyway.

File: ShinyExpenseTrackerTests/src/ca/ualberta/cs/shinyexpensetracker/test/AddExpenseClaimActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+				ExpenseClaimList claimList = new ExpenseClaimList();
+				claimList.addClaim(expenseClaim);
+				
+				assertTrue("name is not 'URoma Trip'", expenseClaim.getName().toString() == "URoma Trip");
+				assertTrue("startDate is not 'null'", expenseClaim.getStartDate().toString() == "2015-01-01");
+				assertTrue("endDate is not 'null'", expenseClaim.getEndDate().toString() == "2015-01-10");
+				assertTrue("Status is not 'null'", expenseClaim.getStatus() == null);
+				assertTrue("Tag is not 'null'", expenseClaim.getTag() == null);
+				
+				assertTrue("ClaimList is empty", claimList.size() == 1);
```
Comment (ItsTristan) [inline ShinyExpenseTrackerTests/src/ca/ualberta/cs/shinyexpensetracker/test/AddExpenseClaimActivityTest.java]: Can you change these assertTrue's to assertEquals? Java's == operator is really dumb with objects.

**Comment by ItsTristan** [comment]
What's the status on this pull request? Have the changes been made yet?

**Comment by ramish94** [comment]
Yes.

**Comment by ItsTristan** [comment]
* Did you merge it right? ExpenseClaimController has a lot of errors.
  * This merge can't be approved until this is fixed.
  * Delete parts that are redundant and keep the one that makes the most sense.

* Need license headers in your files.
* `testendDate` and `teststartDate` are good. Could use a couple more tests though:
  *  include a test that checks if pressing the done button will add the claim to the claim list (use the `ExpenseClaimController` to make sure it's visible to everyone). For instance:

          assertEquals(0, ClaimController.getClaimList().getCount())
          doneButton.performClick()
          assertEquals(1, ClaimController.getClaimList().getCount())
          // check if the claim that's in the list has the right data

  * maybe include some sanity checks so there aren't any false positives. We've been seeing some weirdness with UI testing stuff; best to be safe:

          assertFalse(endDateDialog.isShowing())
          endDate.performClick()
          assertTrue(endDateDialog.isShowing())

**Comment by ramish94** [comment]
Changed. Take a look and see if it's good

**Comment by ItsTristan** [comment]
Did you run your tests? You can't put asserts in the Main runnable, it'll crash the app when they fail! Change your tests to look like this:

		// Assertions, UI, etc.
		instrumentation.runOnMainSync(new Runnable() {
			@Override
			public void run() {
				// Things that change data only (no asserts here!)
			}
		}
		// Assertions, UI, etc.

Once you do that, fix your tests: they're not correct.

**Comment by ramish94** [comment]
Alright. Take a look now. Just ran the tests and everything passed.

**Pull request merged**